extraConfigIPU.py

```diff
@@ -31,13 +35,53 @@ def is_http_url(url: str) -> bool:
         return False
 
 
-def _redfish_boot_ipu(cc: ClustersConfig, imcs: List[str], iso: str) -> None:
+def connect_with_retry(h: host.Host, username: str, password: str) -> None:
+    retries = 10
+    while retries > 0:
+        try:
+            time.sleep(10)
+            retries -= 1
+            h.ssh_connect(username=username, password=password)
+            return
+        except ssh_exception.NoValidConnectionsError:
+            logger.debug("Failed to connect to IMC, retrying")
+
+# By default the ACC does not have network access on boot, we need a workaround to determine the system is up
+def wait_for_imc_boot(h: host.Host, bmc_user: str, bmc_password: str) -> None:
+    logger.info("Waiting for imc installation to complete")
+
+    # Give time for IMC to settle after triggering reboot
+    time.sleep(30)
+
+    while True:
+        connect_with_retry(h, username=bmc_user, password=bmc_password)
+        time.sleep(5)
+        ret = h.run("cat /work/state")
+        if '0' in ret.out:
+            logger.error("IMC in unexpected state 0 post boot, an error occurred during installation")
+            raise Exception("Error occurred during IMC installation, state was reset to 0")
+        if '2' in ret.out:
+            logger.info("Verify imc has successfully completed installation and is booting into installed media")
+            break
+
+    logger.info("Attempting to ping ACC from IMC")
+
+    while True:
+        connect_with_retry(h, username=bmc_user, password=bmc_password)
+        time.sleep(5)
+        ret = h.run("ping -c 1 192.168.0.2")
+        if ret.returncode == 0:
+            logger.info("ACC is booted, installation complete")
+            return
+
+
+def _redfish_boot_ipu(cc: ClustersConfig, imcs: List[str], iso: str, bmc_user: Optional[str] = "root", bmc_password: Optional[str] = "calvin") -> None:
     def helper(imc: str) -> str:
         logger.info(f"Booting {imc} with {iso_address}")
         bmc = host.BMC.from_bmc(imc)
         bmc.boot_iso_redfish(iso_path=iso_address, retries=5, retry_delay=15)
-        # TODO: We need a way to monitor when the installation is complete
-        # since the acc will not have connectivity on reboot
+        h = host.Host(imc)
+        wait_for_imc_boot(h, bmc_user, bmc_password)
```

Comment:
I think we should assign a fixed ip in the dhcp server and then use that to check if the system is up. Then, it's not IMC specific.

Reply:
That would be much cleaner. But unless we are using the default pkg, there will be connectivity to the ACC over the fastport on boot

Reply:
the new pkg will fix this. I prefer to wait for that and then fix the PR. Then we can merge it.

Reply:
That makes a lot of sense to me, sounds good